### ISSUE TITLE
[Bug]: MilvusException: (code=9, message=quota exceeded[reason=rate type: DMLInsert])

### ISSUE THREAD

**Issue by ghu613** [milvus-io/milvus#33756]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: v2.4.1
- Deployment mode(standalone or cluster): cluster
- MQ type(rocksmq, pulsar or kafka): kafka
- SDK version(e.g. pymilvus v2.0.0rc2): v2.4.1
- OS(Ubuntu or CentOS): RockyLinux
- CPU/Memory: 
- GPU: N/A
- Others:
```

### Current Behavior

Hi there,

Starting from a few days ago, MilvusException: (code=9, message=quota exceeded[reason=rate type: DMLInsert]) errors were observed during Insert operation into a target collection. In `quotaAndLimits` config, only `ttProtection` is enabled and the rest configs are just default values. 

Here's the complete config,

```
quotaAndLimits:
  compactionRate:
    db:
      max: -1
    enabled: false
    max: -1
  ddl:
    collectionRate: -1
    db:
      collectionRate: -1
      partitionRate: -1
    enabled: false
    partitionRate: -1
  dml:
    bulkLoadRate:
      collection:
        max: -1
      db:
        max: -1
      max: -1
      partition:
        max: -1
    deleteRate:
      collection:
        max: -1
      db:
        max: -1
      max: -1
      partition:
        max: -1
    enabled: false
    insertRate:
      collection:
        max: -1
      db:
        max: -1
      max: -1
      partition:
        max: -1
    upsertRate:
      collection:
        max: -1
      db:
        max: -1
      max: -1
      partition:
        max: -1
  dql:
    enabled: false
    queryRate:
      collection:
        max: -1
      db:
        max: -1
      max: -1
      partition:
        max: -1
    searchRate:
      collection:
        max: -1
      db:
        max: -1
      max: -1
      partition:
        max: -1
  enabled: true
  flushRate:
    collection:
      max: -1
    db:
      max: -1
    enabled: false
    max: -1
  indexRate:
    db:
      max: -1
    enabled: false
    max: -1
  limitReading:
    coolOffSpeed: 0.9
    forceDeny: false
    queueProtection:
      enabled: false
      nqInQueueThreshold: -1
      queueLatencyThreshold: -1
    resultProtection:
      enabled: false
      maxReadResultRate: -1
      maxReadResultRatePerCollection: -1
      maxReadResultRatePerDB: -1
  limitWriting:
    diskProtection:
      diskQuota: -1
      diskQuotaPerCollection: -1
      diskQuotaPerDB: -1
      diskQuotaPerPartition: -1
      enabled: true
    forceDeny: false
    growingSegmentsSizeProtection:
      enabled: false
      highWaterLevel: 0.4
      lowWaterLevel: 0.2
      minRateRatio: 0.5
    memProtection:
      dataNodeMemoryHighWaterLevel: 0.95
      dataNodeMemoryLowWaterLevel: 0.85
      enabled: true
      queryNodeMemoryHighWaterLevel: 0.95
      queryNodeMemoryLowWaterLevel: 0.85
    ttProtection:
      enabled: true
      maxTimeTickDelay: 300
 ```
**Impact**
At least three days of ingestion tasks are being delayed in Prod environment.

**Debug**
Traced this quota exceeded error in code and it seems like it only happens when the rate limiter limit is [zero](https://github.com/milvus-io/milvus/blob/v2.4.1/internal/util/ratelimitutil/rate_limiter_tree.go#L81) [1].
Meanwhile, rootCoord time tick delay metrics show an old queryNode has a tt delay > 6s [2] which appears to be the root cause but the rootCoord rate limit ratio metric for the target collection is only approximately 0.1%, which means the rate limit factor is ~99.9%. This factor is then [multiplied ](https://github.com/milvus-io/milvus/blob/v2.4.1/internal/rootcoord/quota_center.go#L969-L970) by the current rate limiter limit and becomes the new limit only when it's not Inf [3]. 

**Temporary fix**
mixCoord deployment were restarted but the quota exceeded error still existed. Finally, restarted Proxy deployment and the errors seemed subdued [4].

**Questions**
1. When the rate limit factor is ~99.9%, how did it result in the rate limiter limit at **zero** given no rate limit set in `quotaAndLimits` (all set to -1) [1][3]?
2. Quota center periodically updates the rate in Proxy, but why the issue was still present after mixCoord was restarted? [4]
3. queryCoordMetrics still has the metrics of the old queryNode which seems the metrics was not cleaned properly maybe during rolling update. [2]

**Comment by yanliang567** [comment]
BTW, the fix of #33553 will be included in next release v2.4.5
/assign @ghu613 
/unassign

**Comment by ghu613** [comment]
Appreciate your answer.
1. Still cannot fully understand. The key is when or where the rate limiter limit is set to zero. No rate limiter limit is set on our end. If the rate limit were set to **rate limit factor** * current limit, then it would make sense given the metrics we saw from Prometheus. But zero, is hard to comprehend.
2. After mixCoord was restarted, all tt delay were back within a normal range (~500ms which is not too bad). But Proxy should reflect this because mixCoord is periodically updating Proxy and Proxy should stop rate limiting soon after.
3. Thanks for the fix.

**Comment by SimFG** [comment]
@ghu613 
1. There are actually two values ​​in the settings here, namely the minimum water level and the maximum water level. When the minimum water level is reached, the rate limit factor is a value greater than 0 and less than 1, but when the maximum water level is exceeded, this value will be 0.
2. If it is confirmed that ttdelay has been restored, you can check whether meomey usage exceeds the water level. The default maximum water level setting is 95%

**Comment by ghu613** [comment]
@SimFG Thanks. Just checked, RAM usage of queryNode did increase during that time but not reaching the maximum water level. Also searched in logs, didn't find any such as `"QuotaCenter: QueryNode memory to high water level"` or similar ones for growing segments.

**Comment by SimFG** [comment]
it may be caused by a bug. You can try to use the new version of milvus, the fix is: #33111

**Comment by ghu613** [comment]
> it may be caused by a bug. You can try to use the new version of milvus, the fix is: #33111

Will do, thanks!
